File: 2024_AoC/day02/day02_part1.py
def safe(nbs):
  incr = True
  if (nbs[1] < nbs[0]):
    incr = False
  for i in range(len(parsed) - 1):
    diff = nbs[i+1] - nbs[i]
    if (incr and diff >=1 and diff <=3):
      continue
    elif (not incr and diff <=-1 and diff >=-3):
      continue
    else:
      return False
  return True


f = open("input.txt", "r")
total_safe = 0

for line in f:
  parsed = line.split(" ")
  nbs = map(int, parsed)
  total_safe += safe(list(nbs))
print(total_safe)

File: 2015_AoC/day02/day02_part1.go
package main

import (
  "fmt"
  "os"
  "strings"
)

const fileName = "input.txt"

func calculate_wrapping_paper(src string) int {
  var l, w, h int

  fmt.Sscanf(src, "%dx%dx%d", &l, &w, &h)

  s1 := l * w
  s2 := l * h
  s3 := w * h

  wrapping_paper := 2*s1 + 2*s2 + 2*s3 + min(s1, s2, s3)
  return wrapping_paper
}

func main() {
  data, err := os.ReadFile(fileName)
  if err != nil {
    panic(err)
  }
  lines := strings.Split(string(data), "\n")
  result := 0
  for _, line := range lines {
    result += calculate_wrapping_paper(line)
  }
  fmt.Println(result)
}
